test/asyncprogressworker.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const buildType = process.config.target_defaults.default_configuration;
+const assert = require('assert');
+
+test(require(`./build/${buildType}/binding.node`));
+test(require(`./build/${buildType}/binding_noexcept.node`));
+
+function test(binding) {
+  success(binding);
+  fail(binding);
+  return;
+}
+
+function success(binding) {
+  let progress = 0
+  binding.asyncprogressworker.doWork(true, {}, (err, _progress) => {
+    if (err) {
+      assert.fail(err);
+      return;
+    }
+    progress = _progress;
+  });
+  process.on('exit', () => {
```

Comment:
I was trying to add this case. But one problem is AsyncProgressWorker doesn't save every progress to call into JS. It could be overridden by design. So there is no guarantee on exact callback. 

Side note: AsyncProgressQueueWorker would save every progress to callback.

Reply:
Ok, understood.  Is there any way to add to the test so that it validates that we were called with one of the expected values?  Right now we validate that the callback was called, but nothing to make sure the parameter being passed is called, right?

Reply:
We could synchronize between the main thread and the worker, make sure each callback was successfully invoked. I'd add this case ASAP.

Reply:
I've updated the test case to ensure each process report should be handled properly in JS land :)